sklearn/preprocessing/tests/test_data.py

```diff
@@ -1535,9 +1528,17 @@ def _check_one_hot(X, X2, cat, n_features):
     assert_array_equal(toarray(B), toarray(D))
 
 
+def test_one_hot_encoder_string():
+    X = [['cat', 'domestic'], ['wolf', 'wild']]
+    enc = OneHotEncoder()
+    enc.fit(X)
+    Xtr = enc.transform([['cat', 'wild']])
+    assert_array_equal(toarray(Xtr), [[1, 0, 0, 1]])
+
+
 def test_one_hot_encoder_categorical_features():
     X = np.array([[3, 2, 1], [0, 1, 1]])
-    X2 = np.array([[1, 1, 1]])
+    X2 = np.array([[3, 1, 1]])
```

Comment:
Ok. It is clear to me what you expect. What do you think is the best way to go about this ? And an if/else clause and revert to the old logic when the array is int ? Or keep processing the array as `object` type and emulate the old functionality by using `np.arange` arrays.